src/controllers/cart.js

```diff
@@ -41,29 +41,27 @@ exports.postCartItemByDishId = async (req, res, next) => {
 
     let cartItem = await Cart.findOne({
       customerId: req.user.id,
-      'cartItems.dishId': req.params.id,
+      'cartItems.dishId': req.body.id,
     });
     if (cartItem) return next(new CustomError('item already in cart', 300));
 
     cartItem = {
-      dishId: req.params.id,
+      dishId: req.body.id,
```

Comment:
How are you adding a dish to the cart without checking if that dish exists?
you have to check if the dishId is exists